js/geo/transform.js

```diff
@@ -186,16 +179,16 @@ class Transform {
     zoomScale(zoom) { return Math.pow(2, zoom); }
     scaleZoom(scale) { return Math.log(scale) / Math.LN2; }
 
-    project(lnglat, worldSize) {
```

Comment:
> It does appear that worldSize has some relation to tileSize.

This `tileSize` (which I mentioned in my comment above) is a global constant for `mapbox-gl` and does not need parametrization.